tensorflow/core/kernels/tile_functor_gpu.cu.cc

```diff
@@ -66,8 +66,8 @@ void TileSimple(const Device& d, Tensor* out, const Tensor& in) {
   // therefore we are doing a sync copy effectively.
   d.memcpyHostToDevice(dev_buf, host_buf.data(), num_bytes);
   // Launch kernel to q[...] = p[...].
-  const T* p = reinterpret_cast<const T*>(in.flat<T>().data());
-  T* q = reinterpret_cast<T*>(const_cast<T*>((out->flat<T>().data())));
+  const T* p = in.flat<T>().data();
+  T* q = const_cast<T*>((out->flat<T>().data()));
```

Comment:
You keep trying to leave casts in place.  Why?  The `const_cast` should not be necessary.